Test improvements, add verifier-test.sh to ginkgo

joestringer opened the pull request:
Add the verifier test script to the regular ginkgo testsuite. If a datapath change is likely to cause verifier complexity issues, then this specific test should catch it.

joestringer committed:
test: Add bpf/verifier-test.sh to ginkgo

Add a basic test target that attempts to load all of the built BPF
programs into the ginkgo VM's kernel. This is useful for validating, for
instance, that pending datapath changes will not cause complexity issues
on the older 4.9 kernel used by the standard ginkgo image.

Signed-off-by: Joe Stringer <joe@covalent.io>

ianvernon commented on test/config/config.go:
```diff
@@ -38,4 +39,6 @@ func (c *CiliumTestConfigType) ParseFlags() {
 		"Specify a custom command to fetch SSH configuration (eg: 'vagrant ssh-config')")
 	flag.BoolVar(&c.ShowCommands, "cilium.showCommands", false,
 		"Output which commands are ran to stdout")
+	flag.BoolVar(&c.GatherLogs, "cilium.gatherLogs", true,
```
Update documentation in `Documentation/contributing.rst` to reflect this new command.

joestringer replied:
done.

ianvernon commented on test/runtime/verifier.go:
```diff
@@ -0,0 +1,59 @@
+		Expect(err).Should(BeNil())
+	})
+
+	FIt("Runs verifier test script", func() {
```
FIt --> It ?

joestringer replied:
done.

ianvernon commented on test/runtime/verifier.go:
```diff
@@ -0,0 +1,59 @@
+	})
+
+	AfterAll(func() {
+		err := vm.RestartCilium()
```
Add an `AfterEach` which gathers logs like the other runtime tests do.

joestringer replied:
done.

ianvernon commented on test/runtime/ct.go:
```diff
@@ -507,7 +507,7 @@ var _ = Describe("DisabledRuntimeValidatedConntrackTable", func() {
 		By("Removing policy `l3-l4-policy-server-3`")
 		res = vm.PolicyDel("l3-l4-policy-server-3")
-		res.WasSuccessful()
+		res.ExpectSuccess()
```
Good catch!

ianvernon commented on test/runtime/verifier.go:
```diff
@@ -0,0 +1,66 @@
+)
+
+const (
+	verifierTest = "RuntimeVerifier"
```
`RuntimeVerifier` --> `RuntimeValidatedVerifier`.

joestringer replied:
What's the purpose of the "Validated" portion of the name at this point? I thought thit was just a temporary thing while we migrated from the old bash tests.

joestringer committed:
test: Add failure reporting without Cilium

An upcoming test deliberately disables Cilium and performs some tests.
If there is a failure during the test, on failure it should not attempt
to fetch logs from Cilium because it's not running.

Signed-off-by: Joe Stringer <joe@covalent.io>
